libsolidity/ExpressionCompiler.cpp

```diff
@@ -489,19 +489,36 @@ void ExpressionCompiler::endVisit(MemberAccess const& _memberAccess)
 		m_currentLValue.retrieveValueIfLValueNotRequested(_memberAccess);
 		break;
 	}
+	case Type::Category::Enum:
+	{
+		EnumType const& type = dynamic_cast<EnumType const&>(*_memberAccess.getExpression().getType());
+		EnumDefinition const& enumDef = type.getEnumDefinition();
+		m_context << enumDef.getMemberValue(_memberAccess.getMemberName());
+		break;
+	}
 	case Type::Category::TypeType:
 	{
 		TypeType const& type = dynamic_cast<TypeType const&>(*_memberAccess.getExpression().getType());
+		ContractType const* contractType;
+		EnumType const* enumType;
 		if (type.getMembers().getMemberType(member))
 		{
-			ContractDefinition const& contract = dynamic_cast<ContractType const&>(*type.getActualType())
-													.getContractDefinition();
-			for (ASTPointer<FunctionDefinition> const& function: contract.getDefinedFunctions())
-				if (function->getName() == member)
-				{
-					m_context << m_context.getFunctionEntryLabel(*function).pushTag();
-					return;
-				}
+			if ((contractType = dynamic_cast<ContractType const*>(type.getActualType().get())))
```

Comment:
I see what you are trying to save by suggesting this but it does not seem to compile. A pity. Would have been nice to avoid having to declare type of contractType (and later enumType) above.